corehq/extensions/interface.py

```diff
@@ -0,0 +1,113 @@
+import importlib
+import inspect
+
+from dimagi.utils.logging import notify_exception
+
+
+class ExtensionError(Exception):
+    pass
+
+
+class Extension:
+    def __init__(self, point, callable_ref, domains):
+        self.point = point
+        self.callable = callable_ref
+        self.domains = set(domains) if domains else None
+
+    def validate(self, expected_args):
+        spec = inspect.getfullargspec(self.callable)
+        unconsumed_args = set(expected_args) - set(spec.args)
+        if unconsumed_args and not spec.varkw:
+            raise ExtensionError(f"Not all extension point args are consumed: {unconsumed_args}")
+
+    def call_for_domain(self, domain):
```

Comment:
This does not call anything. I think the intent was clearer with a `should_` prefix.
```suggestion
    def should_call_for_domain(self, domain):
```